File: 636-ExclusiveTimeOfFunctions/636-ExclusiveTimeOfFunctions.java
// Last updated: 6/10/2026, 7:24:14 PM
import java.util.*;
class Solution {
    public int[] exclusiveTime(int n, List<String> logs) {
        int ans[] = new int[n];
        Arrays.fill(ans , 0);

        Stack<Integer> stack = new Stack<>();
        int prev = 0;
        for(String log : logs) {
            String[] parts = log.split(":");
            int id = Integer.parseInt(parts[0]);
            int time = Integer.parseInt(parts[2]);
            String op = parts[1];

            if(op.equals("start")) {
                if(!stack.empty()) {
                    ans[stack.peek()] += time - prev;
                }
                stack.push(id);
                prev = time;
            } else {
                ans[stack.pop()] += time - prev + 1;
                prev = time+1;
            }
        }
        return ans;
    }
}

File: 636-ExclusiveTimeOfFunctions/636-ExclusiveTimeOfFunctions.cpp
// Last updated: 6/10/2026, 7:24:19 PM
class Solution {
public:
    vector<int> exclusiveTime(int n, vector<string>& logs) {
        vector<int> times(n, 0);
        stack<int> st;
        int prev_time = 0;
        
        for (auto& log : logs) {
            int id, time;
            string type;
            parseLog(log, id, type, time);
            
            if (type == "start") {
                if (!st.empty())
                    times[st.top()] += time - prev_time;  // running function till now
                st.push(id);
                prev_time = time;
            } else {
                times[st.top()] += time - prev_time + 1;  // inclusive end
                st.pop();
                prev_time = time + 1; // next function resumes after current ends
            }
        }
        return times;
    }
    
private:
    void parseLog(const string& log, int& id, string& type, int& time) {
        int pos1 = log.find(':'), pos2 = log.find(':', pos1 + 1);
        id = stoi(log.substr(0, pos1));
        type = log.substr(pos1 + 1, pos2 - pos1 - 1);
        time = stoi(log.substr(pos2 + 1));
    }
};